include/deal.II/base/utilities.h

```diff
@@ -72,6 +75,32 @@ namespace Utilities
   std::string
   dealii_version_string();
 
+  /**
+   * Assign to each point in @p points an index using Hilbert space filling curve.
+   * To that end, bounding box for @p points will be determined, based on which their
+   * integer coordinates are calculated.
+   * The linear index is given as a dim-collection of bits, from high to low.
+   *
+   * The depth of the Hilbert curve (i.e. the number of bits per dimension) by
+   * default is equal to <code>std::numeric_limits<Integer>::digits/code>. The
+   * function is instantiated for <code>unsigned long int</code> and
+   * <code>double</code>.
+   */
+  template <typename Integer, int dim, typename Number>
+  std::vector<std::array<Integer, dim>>
```

Comment:
I think you might as well unconditionally use `std::uint64_t` here. That should be good enough for most anything, but also not be wasteful.

Reply:
I like the fact that it's general. IMO there is no reason not to make it so.

Reply:
But you are right, that 64-bit is a typical choice for these applications. Note that we would have to limit the actual bits to 52 at least in `double` version of the function.

Reply:
The thing with being general is that it requires to write down a lot of template instantiations by hand. What do you gain here if you keep it generic?

Reply:
i switched to `std::uint64_t`.